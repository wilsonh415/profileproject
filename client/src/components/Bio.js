import React from 'react';
import { Container, Row } from 'react-bootstrap';
import AvatarEditor from 'react-avatar-editor';
// import profilepic from '../images/profilepic.JPG';
import profilepic from '../images/foto.JPG';
import gmailicon from '../images/gmailicon.png';
import githubicon from '../images/githubicon.svg';
import linkedinicon from '../images/linkedinicon.png';
 
class Bio extends React.Component {
 
   render() {
       const bioText = {
           margin: "10px",
           color: "black",
           fontFamily: "Georgia",
           fontSize: "16px"
       };
       const linkStyles = {
           marginTop: "2vh",
           marginBottom: "8vh"
       };
       return (
           <div className="profileSidebar">
               <Container>
                   <Row>
                       <div className="profilePic">
                           <AvatarEditor
                           width={210}
                           height={210}
                           scale={1.0}
                           border={0}
                           style={{borderRadius: "50%"}}
                           image={profilepic}/>
                       </div>
                   </Row>
                   <h3 style={{fontFamily: "Georgia"}}>Wilson Huang</h3>
                   <div style={linkStyles}>
                       <a href="mailto:wilsonh415@gmail.com">
                           <img width="30" height="30" style={{margin: "10px", marginRight: "20px"}}
                           src={gmailicon} alt="Gmail Icon"></img>
                       </a>
                       <a href="https://github.com/wilsonh415" 
                       target="_blank" rel="noopener noreferrer">
                           <img width="30" height="30" style={{margin: "10px", marginRight: "20px"}}
                           src={githubicon} alt="Github Icon"></img>
                       </a>
                       <a href="https://www.linkedin.com/in/wilsonhuang5/" 
                       target="_blank" rel="noopener noreferrer">
                           <img width="30" height="30" style={{margin: "10px"}}
                           src={linkedinicon} alt="LinkedIn Icon"></img>
                       </a>
                   </div>
                   <Row>
                       <p style={bioText}>
                           I'm a 4th year computer science student at
                           Cal Poly, SLO graduating in Spring 2021.
                           Born, raised, and currently living in San Francisco.
                           I love running, biking, exploring places, and
                           learning new perspectives! Looking for full-time
                           employment developing software.
                       </p>
                   </Row>
               </Container>
           </div>
       )
   }
}
 
export default Bio;
