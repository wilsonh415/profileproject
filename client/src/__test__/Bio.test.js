import React from 'react';
import Bio from '../components/Bio';
import { shallow } from 'enzyme';
import { configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

test("Bio component renders properly", () => {
    const component = shallow(<Bio></Bio>);
    expect(JSON.stringify(component)).toMatchSnapshot();
});

test('Name in Bio component is correct', () => {
    const component = shallow(<Bio></Bio>);
    const name = component.find("h3").text();
    expect(name).toEqual("Wilson Huang");
});
