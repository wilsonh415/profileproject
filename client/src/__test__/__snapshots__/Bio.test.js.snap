// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio component renders properly 1`] = `"{}"`;
